### PR TITLE
[JFR] add support for opto object allocations sampling

### PR DESCRIPTION
**Pull request by D-D-H**
Summary: add support for opto object allocations sampling

Test Plan: test/jdk/jdk/jfr/event/objectsprofiling/TestOptoObjectAllocationsSampling.java

Reviewers: kuaiwei, luchsh, sanhong

Issue: #7

### CODE REVIEW

File: src/hotspot/share/jfr/objectprofiler/objectProfiler.hpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#ifndef SHARE_VM_JFR_OBJECTROFILER_OBJECTPROFILER_HPP
```
Comment (kuaiwei) [inline src/hotspot/share/jfr/objectprofiler/objectProfiler.hpp]: add copyright header

Comment (D-D-H) [reply src/hotspot/share/jfr/objectprofiler/objectProfiler.hpp]: fixed

File: src/hotspot/share/jfr/objectprofiler/objectProfiler.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
+#include "runtime/vmThread.hpp"
+#include "jfr/objectprofiler/objectProfiler.hpp"
+#include "jfr/utilities/jfrTryLock.hpp"
+#include "jfr/jfrEvents.hpp"
+
+jint ObjectProfiler::_enabled = JNI_FALSE;
```
[the remark this answers is not in the document]
Comment (D-D-H) [reply src/hotspot/share/jfr/objectprofiler/objectProfiler.cpp]: fixed

File: src/hotspot/share/jfr/objectprofiler/objectProfiler.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#include "runtime/vmThread.hpp"
```
[the remark this answers is not in the document]
Comment (D-D-H) [reply src/hotspot/share/jfr/objectprofiler/objectProfiler.cpp]: fixed

**Pull request merged**